resolver.go

```diff
@@ -140,38 +142,67 @@ func (r *resolver) Start() error {
 	if r.err != nil {
 		return r.err
 	}
+	// startCh is to seralize resolver Start and Stop
+	r.Lock()
+	r.startCh = make(chan struct{})
```

Comment:
Can we initialize the channel in the constructor `NewResolver()` ?
That way I think we can avoid adding the mutex.